VideoWeb/VideoWeb.Common/Caching/IConferenceCache.cs

```diff
@@ -7,7 +7,8 @@ namespace VideoWeb.Common.Caching
 {
     public interface IConferenceCache
     {
-        Task AddConferenceToCache(ConferenceDetailsResponse conferenceResponse);
-        Conference GetConference(Guid id);
+        Task AddConferenceAsync(ConferenceDetailsResponse conferenceResponse);
+        Task <Conference>GetOrAddConferenceAsync(Guid id, Func<Task<ConferenceDetailsResponse>> addConferenceDetailsFactory);
+        Task<Conference> GetConferenceAsync(Guid id);
```

Comment:
If we have an GetOrAddConferenceAsync, we can get rid of the GetConferenceAsync, both methods achieve the same thing

Reply:
This should be a separate PR

Reply:
But yes, we should use this where ever we are getting from cache and if null, setting it